qiskit_nature/transformers/base_transformer.py

```diff
@@ -18,14 +18,15 @@
 
 
 class BaseTransformer(ABC):
-    """The interface for implementing methods which map from one `QMolecule` to another. These
-    methods may or may not affect the size of the Hilbert space.
+    """The interface for implementing methods which map from one `QMolecule` or
+    'WatsonHamiltonian' to another. These methods may or may not affect the size of the Hilbert
+    space.
     """
 
     @abstractmethod
     def transform(self, q_molecule: QMolecule) -> QMolecule:
```

Comment:
Since you want to add the `WatsonHamiltonian` to this interface, it should also go in the type hint here. Although that will likely lead to `mypy` issues, too..

Reply:
That was my initial try but mypy issues, exactly.

Reply:
You could add a `# type: ignore` comment in these cases just to get `mypy` to shut up for now, until we refactor the drivers and `QMolecule`/`WatsonHamiltonian` classes.